rllib/examples/rl_modules/classes/modelv2_to_rlm.py

```diff
@@ -0,0 +1,104 @@
+from typing import Any, Dict
+
+from ray.rllib.core.columns import Columns
+from ray.rllib.core.rl_module.apis import ValueFunctionAPI
+from ray.rllib.core.rl_module.torch import TorchRLModule
+from ray.rllib.models.torch.torch_distributions import (
+    TorchCategorical,
+    TorchDiagGaussian,
+    TorchMultiCategorical,
+    TorchMultiDistribution,
+    TorchSquashedGaussian,
+)
+from ray.rllib.models.torch.torch_action_dist import (
+    TorchCategorical as OldTorchCategorical,
+    TorchDiagGaussian as OldTorchDiagGaussian,
+    TorchMultiActionDistribution as OldTorchMultiActionDistribution,
+    TorchMultiCategorical as OldTorchMultiCategorical,
+    TorchSquashedGaussian as OldTorchSquashedGaussian,
+)
+from ray.rllib.policy.torch_policy_v2 import TorchPolicyV2
+from ray.rllib.utils.annotations import override
+
+
+class ModelV2ToRLModule(TorchRLModule, ValueFunctionAPI):
+    """An RLModule containing a (old stack) ModelV2, provided by a policy checkpoint."""
+
+    @override(TorchRLModule)
+    def setup(self):
+        super().setup()
+
+        # Get the policy checkpoint from the `model_config_dict`.
+        policy_checkpoint_dir = self.config.model_config_dict.get(
+            "policy_checkpoint_dir"
+        )
+        if policy_checkpoint_dir is None:
+            raise ValueError(
+                "The `model_config_dict` of your RLModule must contain a "
+                "`policy_checkpoint_dir` key pointing to the policy checkpoint "
+                "directory! You can find this dir under the Algorithm's checkpoint dir "
+                "in subdirectory: [algo checkpoint dir]/policies/[policy ID, e.g. "
+                "`default_policy`]."
+            )
+
+        # Create a temporary policy object.
+        policy = TorchPolicyV2.from_checkpoint(policy_checkpoint_dir)
+        self._model_v2 = policy.model
+
+        # Translate the action dist classes from the old API stack to the new.
+        self._action_dist_class = self._translate_dist_class(policy.dist_class)
+
+        # Erase the torch policy from memory, so it can be garbage collected.
+        del policy
+
+    def _forward_inference(self, batch: Dict[str, Any], **kwargs) -> Dict[str, Any]:
+        nn_output, state_out = self._model_v2(batch)
```

Comment:
Nice!!! So simple. I wonder if this works fine with any more complicated setup like LSTM or MARWIL

Reply:
Probably not :| but it's just a start ...

Reply:
I have a LSTM example script coming up in another PR (the one that adds the wrapper "from config").